include/orcania.h

```diff
@@ -10,113 +10,141 @@
 #ifndef __ORCANIA_H__
 #define __ORCANIA_H__
 
-#ifndef U_DISABLE_JANSSON
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+#ifndef O_EXTERN
+# ifdef O_SHARED_LIBRARY
+#  ifdef _WIN32
+#   define O_EXTERN __declspec(dllexport)
+#  else
+#   define O_EXTERN extern
+#  endif
+# else
+#  ifdef O_STATIC_LIBRARY
+#   define O_EXTERN
+#  else
+#   define O_EXTERN extern
+#  endif
+# endif
+#endif
+
+#ifndef ORCANIA_VERSION
+#define ORCANIA_VERSION LIBRARY_VERSION
+#endif
+
+#ifdef USE_JANSSON
 #include <jansson.h>
 #endif
 
-#define ORCANIA_VERSION 1.1.1
+/**
+ * Return the library version.
+ */
+O_EXTERN const char *o_version();
+
 /**
  * char * str_replace(const char * source, char * old, char * new)
  * replace all occurences of old by new in the string source
  * return a char * with the new value
  * return NULL on error
  * returned value must be free'd after use
  */
-char * str_replace(const char * source, const char * str_old, const char * str_new);
+O_EXTERN char * str_replace(const char * source, const char * str_old, const char * str_new);
```

Comment:
I don't like having the macro `O_EXTERN` prefixing every function provided by Orcania, is there a reason to do this?